apps/prairielearn/elements/pl-external-grader-variables/pl-external-grader-variables.py

```diff
@@ -28,7 +28,7 @@ def prepare(element_html: str, data: pl.QuestionData) -> None:
 
             html_variables.append(var_dict)
 
-        elif child.tag is lxml.etree.Comment:
+        elif child.tag is lxml.etree.Comment:  # pyright: ignore[reportUnnecessaryComparison]
```

Comment:
The typechecker isn't wrong here, the types are. With type errors, I typically like to file an issue upstream and then link to it whenever we opt to ignore errors, e.g.:

```suggestion
        # https://github.com/lxml/lxml-stubs/issues/...
        elif child.tag is lxml.etree.Comment:  # pyright: ignore[reportUnnecessaryComparison]
```

Reply:
I opened an issue here: https://github.com/lxml/lxml-stubs/issues/102

Reply:
Oops, turns out we actually use https://github.com/abelcheung/types-lxml 🙈 

https://github.com/PrairieLearn/PrairieLearn/blob/a768f5dd12cd2c4b9936eb0c7384a55cf0cc8089/images/plbase/python-requirements.txt#L38

Reply:
@nwalters512 They suggest changing the check: https://github.com/abelcheung/types-lxml/issues/46

```
for el in root:
    if isinstance(el, etree._Comment):
        comment.append(el.text)
```

does that work for you?

Reply:
[el oh el](https://github.com/abelcheung/types-lxml/blob/3370d147e651a003cc77dc2d6b7ff701e9aa738e/src/lxml-stubs/etree/_element.pyi#L440-L446)

```py
# Mypy: Liskov!
# Lxml: No Liskov!
# Mypy: I am *THE* authority here!
# Lxml: I will *NEVER* submit to you! Fuck off!
# Mypy: Now die!
#
# So here we are.
```

This who API just seems extremely ******. `Element` is a factory function, `_Element` is the class itself? 🙃 

Yep, `isinstance(el, _Comment)` seems like the way to go, unfortunately.

Reply:
Resolved in 11aacebf6f2461578e5a1f7b9d99c6d8d397bdef.